src/pages/index.js

```diff
@@ -4,90 +4,185 @@ import PopupWithForm from "../components/PopupWithForm.js";
-      job: data.description,
-    });
+    return api
+      .updateUserInfo({
+        name: data.name,
+        about: data.description,
+      })
+      .then((updatedUser) => {
+        userInfo.setUserInfo({
+          name: updatedUser.name,
+          job: updatedUser.about,
+        });
+      });
   },
   formValidators["profile-form"]
 );
 profilePopup.setEventListeners();
 
-// Add card popup
+// ✅ Add card popup (Fix: Only adds new card)
 const addCardPopup = new PopupWithForm(
   ".modal_type_add-card",
   (data) => {
-    const newCard = createCard({ name: data.placeName, link: data.imageLink });
-    cardSection.addItem(newCard); // Add the new card to the DOM
-    if (formValidators["add-card"]) {
-      formValidators["add-card"].disableButton(); // Disable the button after submission
-    } else {
-      console.error('Validator for "add-card" is not defined');
-    }
+    return api
+      .addCard({ name: data.placeName, link: data.imageLink })
+      .then((newCard) => {
+        if (!newCard || !newCard._id) {
+          throw new Error("Error: New card does not have a valid ID.");
+        }
+        const cardElement = createCard(newCard);
+        if (cardElement) cardSection.addItem(cardElement);
+      });
   },
   formValidators["add-card"]
 );
 addCardPopup.setEventListeners();
 
-// Event listeners for popups
+// ✅ Avatar update popup (Fix: Use userInfo for setting)
+const avatarPopup = new PopupWithForm(
+  ".modal_type_avatar",
+  (data) => {
+    return api.updateUserAvatar({ avatar: data.avatar }).then((res) => {
+      userInfo.setUserInfo({ avatar: res.avatar });
+    });
+  },
+  formValidators["avatar-form"]
+);
+avatarPopup.setEventListeners();
+
+// ✅ Open avatar popup on button click
+document.querySelector(".avatar__edit-button").addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// ✅ Delete confirmation popup
+let cardToDelete = null;
+let cardIdToDelete = null;
+
+const deletePopup = new PopupWithForm(".modal_type_delete", () => {
+  if (cardToDelete && cardIdToDelete) {
+    return api.deleteCard(cardIdToDelete).then(() => {
+      cardToDelete.deleteCard();
+      cardToDelete = null;
+      cardIdToDelete = null;
+      deletePopup.close();
+    });
+  }
+  return Promise.reject("No card selected for deletion");
+});
+deletePopup.setEventListeners();
+
+// ✅ Function to open delete popup
+function openDeletePopup(cardId, cardInstance) {
+  if (!cardId) {
+    console.error("Error: Trying to delete a card without an ID");
+    return;
+  }
+  cardIdToDelete = cardId;
+  cardToDelete = cardInstance;
+  deletePopup.open();
+}
+
+// ✅ Event listeners for profile popups
 document
   .querySelector(".profile__edit-button")
   .addEventListener("click", () => {
     const userData = userInfo.getUserInfo();
-    document.forms["profile-form"].name.value = userData.name;
-    document.forms["profile-form"].description.value = userData.job;
+    profilePopup.setInputValues(userData);
```

Comment:
* There is no `job` text in the `job` input when I open the `profile` modal https://snipboard.io/Na1qPt.jpg . Please, fill it correctly

You should update the object with `description` because this value is in the `name` attribute of the `job` input

```js

    const userData = userInfo.getUserInfo();
    userData.description = userData.job;  // update the object
    profilePopup.setInputValues(userData);
```